src/odemis/acq/stitching/__init__.py

```diff
@@ -16,6 +16,62 @@
 '''
 from __future__ import division
 
-
 from ._registrar import *
 from ._weaver import *
+
+import copy
+
+
+def register(tiles, method="REGISTER_SHIFT"):
+    """ 
+    tiles (list of DataArray of shape YX or tuples of DataArrays): The tiles to compute the registration. 
+    If it's tuples, the first tile of each tuple is the “main tile”, and the following ones are 
+    dependent tiles.
+    method (REGISTER_*): REGISTER_SHIFT → ShiftRegistrar, REGISTER_IDENTITY → IdentityRegistrar
+    returns:
+        tiles (list of DataArray of shape YX or tuples of DataArrays): The tiles as passed, but with updated 
+        MD_POS metadata
+    """
+    
+    if method == "REGISTER_SHIFT":
+        registrar = ShiftRegistrar()
+        
+    elif method == "REGISTER_IDENTITY":
+        registrar = IdentityRegistrar()
+    
+    else:
+        raise ValueError("Invalid registrar") 
+
+    updatedTiles=[]
+    for i in range(len(tiles)):
+        # Separate tile and dependent_tiles
+        if isinstance(tiles[i],tuple):
+            tile = tiles[i][0]
+            dep_tiles = tiles[i][1:]
+        else:
+            tile = tiles[i]
+            dep_tiles = None
+
+        registrar.addTile(tile,dep_tiles)
+        
```

Comment:
The algorithm needs to be slightly modified: first, add all the tiles, and only once _all_ the tiles have been received by the registrar, get the positions, and updates the data.

In the current form, it might work because the two registrars are "greedy", and the position of a new tile is immediately computed and never updated. However, new registrars could well update the position of previous tiles, when new information is provided.